src/VisualStudio/Core/Def/DesignerAttribute/VisualStudioDesignerAttributeService.cs

```diff
@@ -106,6 +106,11 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
 
         private async ValueTask ProcessWorkspaceChangeAsync(CancellationToken cancellationToken)
         {
+            var statusService = _workspace.Services.GetRequiredService<IWorkspaceStatusService>();
+            await statusService.WaitUntilFullyLoadedAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
This is reasonably cheap, and won't result in us continually doing this wait and then cancelling it again?

Reply:
yes.